utils/src/treev2/quickPickWizard/GenericQuickPickStep.ts

```diff
@@ -0,0 +1,112 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import * as types from '../../../index';
+import * as vscode from 'vscode';
+import { getLastNode } from './QuickPickWizardContext';
+import { AzureWizardPromptStep } from '../../wizard/AzureWizardPromptStep';
+import { NoResourceFoundError } from '../../errors';
+import { parseError } from '../../parseError';
+
+export interface GenericQuickPickOptions {
+    skipIfOne?: boolean;
+}
+
+export interface SkipIfOneQuickPickOptions extends GenericQuickPickOptions {
+    skipIfOne?: true;
+}
+
+export abstract class GenericQuickPickStep<TNode extends unknown, TContext extends types.QuickPickWizardContext<TNode>, TOptions extends GenericQuickPickOptions> extends AzureWizardPromptStep<TContext> {
+    public readonly supportsDuplicateSteps = true;
+
+    public constructor(
+        protected readonly treeDataProvider: vscode.TreeDataProvider<TNode>,
+        protected readonly pickOptions: TOptions
+    ) {
+        super();
+    }
+
+    public async prompt(wizardContext: TContext): Promise<void> {
+        try {
+            const pick = await this.promptInternal(wizardContext);
+            wizardContext.pickedNodes.push(pick);
+        } catch (err) {
+            const error = parseError(err);
+            if (error.errorType === 'GoBackError') {
+                // Instead of wiping out a property value, which is the default wizard behavior for `GoBackError`, pop the most recent
+                // value off from the provenance of the picks
+                wizardContext.pickedNodes.pop();

```

Comment:
Going back currently doesn't work if a previous step was skipped. Skipped here meaning there was only one pick which was auto-selected and pushed into `pickedNodes` and never shows a prompt.

If a step never shows a prompt, the wizard will pop it from the finished steps array. Which in our case is bad because then the steps and pickedNodes become out of sync.

That happens here: https://github.com/microsoft/vscode-azuretools/blob/4f1923e6cff24fc23987f4b139cdf23f5aa4d716/utils/src/wizard/AzureWizard.ts#L232

Example:

Before hitting the back button:

```
currentStep: PickAppResourceStep
finishedSteps: [PickSubscriptionStep, PickGroupStep]
pickedNodes: [SubscriptionItem, ResourceTypeItem]
```

After hitting the back button:

```
currentStep: PickSubscriptionStep
finishedSteps: [PickSubscriptionStep]
pickedNodes: [SubscriptionItem]
```

Now the picks shown in the PickSubscriptionStep are children of the SubscriptionItem, which is no good.